File: skills/revelio-python/scripts/wrds_connection_template.py
"""Safe WRDS connection template for Revelio Labs queries.

Credentials are read from environment variables:
    WRDS_USERNAME
    WRDS_PASSWORD
"""

from __future__ import annotations

import os

from sqlalchemy import create_engine, text


def get_wrds_engine():
    username = os.environ.get("WRDS_USERNAME")
    password = os.environ.get("WRDS_PASSWORD")

    if not username or not password:
        raise RuntimeError(
            "Set WRDS_USERNAME and WRDS_PASSWORD environment variables before connecting."
        )

    url = (
        "postgresql+psycopg2://"
        f"{username}:{password}"
        "@wrds-pgdata.wharton.upenn.edu:9737/wrds?sslmode=require"
    )
    return create_engine(url)


def example_query(limit: int = 10):
    engine = get_wrds_engine()
    query = text(
        """
        SELECT user_id, user_country, highest_degree, updated_dt
        FROM revelio.individual_user
        WHERE user_country = :country
        LIMIT :limit
        """
    )
    with engine.connect() as conn:
        return conn.execute(query, {"country": "Morocco", "limit": limit}).fetchall()


if __name__ == "__main__":
    rows = example_query()
    print(f"Fetched {len(rows)} rows.")

File: skills/revelio-r/scripts/wrds_connection_template.R
# Safe WRDS connection template for Revelio Labs queries.
#
# Credentials are read from environment variables:
#   WRDS_USERNAME
#   WRDS_PASSWORD

library(DBI)
library(RPostgres)

get_wrds_connection <- function() {
  username <- Sys.getenv("WRDS_USERNAME")
  password <- Sys.getenv("WRDS_PASSWORD")

  if (identical(username, "") || identical(password, "")) {
    stop("Set WRDS_USERNAME and WRDS_PASSWORD environment variables before connecting.")
  }

  dbConnect(
    RPostgres::Postgres(),
    host = "wrds-pgdata.wharton.upenn.edu",
    port = 9737,
    dbname = "wrds",
    sslmode = "require",
    user = username,
    password = password
  )
}

example_query <- function(limit = 10) {
  con <- get_wrds_connection()
  on.exit(dbDisconnect(con), add = TRUE)

  dbGetQuery(
    con,
    paste0(
      "SELECT user_id, user_country, highest_degree, updated_dt ",
      "FROM revelio.individual_user ",
      "WHERE user_country = $1 ",
      "LIMIT $2"
    ),
    params = list("Morocco", limit)
  )
}

